localstack-core/localstack/services/kms/provider.py

```diff
@@ -1078,6 +1078,49 @@ def get_parameters_for_import(
             ParametersValidTo=expiry_date,
         )
 
+    def _decrypt_padded_key_material(
+        import_state: KeyImportState,
+        encrypted_key_material: CiphertextType,
+    ) -> bytes:
+        if import_state.wrapping_algo == AlgorithmSpec.RSAES_PKCS1_V1_5:
+            decrypt_padding = padding.PKCS1v15()
+        elif import_state.wrapping_algo == AlgorithmSpec.RSAES_OAEP_SHA_1:
+            decrypt_padding = padding.OAEP(padding.MGF1(hashes.SHA1()), hashes.SHA1(), None)
+        elif import_state.wrapping_algo == AlgorithmSpec.RSAES_OAEP_SHA_256:
+            decrypt_padding = padding.OAEP(padding.MGF1(hashes.SHA256()), hashes.SHA256(), None)
+        else:
+            raise KMSInvalidStateException(
+                f"Unexpected padding algorithm: '{import_state.wrapping_algo}'"
+            )
+
+        key_material = import_state.key.crypto_key.key.decrypt(
+            encrypted_key_material, decrypt_padding
+        )
+        return key_material
+
+    def _decrypt_wrapped_key_material(
+        import_state: KeyImportState,
+        encrypted_key_material: CiphertextType,
+    ) -> bytes:
+        if import_state.wrapping_algo == AlgorithmSpec.RSA_AES_KEY_WRAP_SHA_1:
+            decrypt_padding = padding.OAEP(padding.MGF1(hashes.SHA1()), hashes.SHA1(), None)
+        elif import_state.wrapping_algo == AlgorithmSpec.RSA_AES_KEY_WRAP_SHA_256:
+            decrypt_padding = padding.OAEP(padding.MGF1(hashes.SHA256()), hashes.SHA256(), None)
+        else:
+            raise KMSInvalidStateException(
+                f"Unexpected wrapping algorithm: '{import_state.wrapping_algo}'"
+            )
+
```

Comment:
As mentioned in the handler, we could just combine a lot of the repetitive parts here 🙂